packages/core/src/lib/light_push/index.ts

```diff
@@ -97,49 +93,69 @@ class LightPush extends BaseProtocol implements ILightPush {
     if (preparationError || !query) {
       return {
         recipients,
-        error: preparationError
+        errors: [preparationError]
       };
     }
 
-    let error: undefined | SendError = undefined;
-    const peer = await this.getPeer(opts?.peerId);
-    const stream = await this.getStream(peer);
+    const peers = await this.getPeers({
+      maxBootstrapPeers: 1,
+      numPeers: this.NUM_PEERS_PROTOCOL
+    });
 
-    try {
-      const res = await pipe(
-        [query.encode()],
-        lp.encode,
-        stream,
-        lp.decode,
-        async (source) => await all(source)
-      );
+    const promises = peers.map(async (peer) => {
+      let error: SendError | undefined;
+      const stream = await this.getStream(peer);
 
       try {
-        const bytes = new Uint8ArrayList();
-        res.forEach((chunk) => {
-          bytes.append(chunk);
-        });
-
-        const response = PushRpc.decode(bytes).response;
-
-        if (response?.isSuccess) {
-          recipients.push(peer.id);
-        } else {
-          log("No response in PushRPC");
-          error = SendError.NO_RPC_RESPONSE;
+        const res = await pipe(
+          [query.encode()],
+          lp.encode,
+          stream,
+          lp.decode,
+          async (source) => await all(source)
+        );
+        try {
+          const bytes = new Uint8ArrayList();
+          res.forEach((chunk) => {
+            bytes.append(chunk);
+          });
+
+          const response = PushRpc.decode(bytes).response;
+
+          if (response?.isSuccess) {
+            recipients.some((recipient) => recipient.equals(peer.id)) ||
+              recipients.push(peer.id);
+          } else {
+            log("No response in PushRPC");
+            error = SendError.NO_RPC_RESPONSE;
+          }
+        } catch (err) {
+          log("Failed to decode push reply", err);
+          error = SendError.DECODE_FAILED;
         }
       } catch (err) {
-        log("Failed to decode push reply", err);
-        error = SendError.DECODE_FAILED;
+        log("Failed to send waku light push request", err);
+        error = SendError.GENERIC_FAIL;
       }
-    } catch (err) {
-      log("Failed to send waku light push request", err);
-      error = SendError.GENERIC_FAIL;
-    }
+
+      return { recipients, error };
+    });
+
+    const results = await Promise.allSettled(promises);
+    const successfulResults = results.filter(
+      (result) => result.status === "fulfilled"
```

Comment:
I guess it never rejects, so we don't need to map and can just cast here

Reply:
I think we should make code explicit enough to understand there are no rejected cases expected in the results (filter can stay just in case)

Reply:
it was more of a pseudocode, good to keep the filter, just the name of `successfulResults` seemed misleading